api/src/opentrons/protocol_api/labware.py

```diff
@@ -98,6 +98,49 @@ def geometry(self) -> WellGeometry:
     def diameter(self) -> Optional[float]:
         return self._geometry.diameter
 
+    @property  # type: ignore
+    @requires_version(2, 9)
+    def length(self) -> Optional[float]:
+        """
+        The length of a well, if the labware has
+        square wells.
+        """
+        return self._geometry._length
+
+    @property  # type: ignore
+    @requires_version(2, 9)
+    def width(self) -> Optional[float]:
+        """
+        The width of a well, if the labware has
+        square wells.
+        """
+        return self._geometry._width
+
+    @property  # type: ignore
+    @requires_version(2, 9)
+    def depth(self) -> float:
+        """
+        The depth of a well in a labware.
+        """
+        return self._geometry._depth
+
+    @property
+    def _depth(self) -> float:
+        MODULE_LOG.warning("This attribute is deprecated."
+                           "Please use 'depth' instead.")
+        return self.depth
```

Comment:
Would it be worth adding comments like this?

```suggestion
    # Preserved for compatibility with workarounds used by Applications Engineering.
    @property
    def _depth(self) -> float:
        MODULE_LOG.warning("This attribute is deprecated."
                           "Please use 'depth' instead.")
        return self.depth
```

Otherwise, I worry we won't know why it's deprecated and what needs to happen before we can safely remove it.